### PR TITLE
Exposure Notification v1.3

### LINKED ISSUE #25: Update Exposure Notification to V1.3
Do a quick update without massive changes because the API is unstable and the new design in v1.3 that uses tokens might be removed in the future.

## Resources
https://static.googleusercontent.com/media/www.google.com/en//covid19/exposurenotifications/pdfs/Android-Exposure-Notification-API-documentation-v1.3.pdf

### PR DESCRIPTION
**Pull request by Apisov**
I tried to minimise changes because there is a chance of different API logic in the next version. 

--- 
Closes #25

### CODE REVIEW

File: app/src/main/java/org/covidwatch/android/data/BaseDao.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
+package org.covidwatch.android.data
+
+import androidx.room.Delete
+import androidx.room.Insert
+import androidx.room.Update
+
+interface BaseDao<T> {
```
Comment (nkumarcc) [inline app/src/main/java/org/covidwatch/android/data/BaseDao.kt:7]: Does this not exist already in Kotlin? Seems like something that should be native

An understand q, not necessarily a fix 😛

**Review by nkumarcc — COMMENTED** [review]
Reviewed like half of this, what's the context of the changes? In particular, how does the `DiagnosisKey`'s implementation differ strongly from what we had before?

File: app/src/main/java/org/covidwatch/android/data/BaseDao.kt (hunk 1 above)
Comment (Apisov) [reply app/src/main/java/org/covidwatch/android/data/BaseDao.kt:7]: I wish but no :)

**Comment by Apisov** [comment]
@nkumarcc 
>Reviewed like half of this, what's the context of the changes? In particular, how does the DiagnosisKey's implementation differ strongly from what we had before?
The main reason is the new version of EN API. Most of the changes do not affect current UI so it should be safe to merge.

We don't have a strong implementation of DiagnosisKeys local or remote source. In general, what we have now is abstract and empty implementations because EN API proven to be uncertain so we better wait until we have a working EN API.

**Pull request merged**